crates/bvs-vault-cw20-tokenized/src/contract.rs

```diff
@@ -0,0 +1,708 @@
+            vault_execute::queue_withdrawal_to(deps, env, info, msg)
+        }
+        CombinedExecuteMsg::RedeemWithdrawalTo(msg) => {
+            // This is the only execute msg that is not passed to the base contract
+            // because it is a custom logic for this vault contract
+            vault_execute::redeem_withdrawal_to(deps, env, info, msg)
+        }
+        _ => {
+            // cw20 compliant messages are passed to the `cw20-base` contract.
+            // Except for the `Burn` and `BurnFrom` messages.
+            receipt_cw20_execute::execute_base(deps, env, info, msg).map_err(Into::into)
+        }
+    }
+}
+
+/// cw20 compliant messages are passed to the `cw20-base` contract.
+/// Except for the `Mint`, `Burn` and `BurnFrom` messages.
+/// The only time receipt token total supply should be changed is through staking and unstaking
+/// More precisely, only through - deposit_for and withdraw_to and redeem_withdrawal_to
+mod receipt_cw20_execute {
+    use cosmwasm_std::{Addr, StdError, StdResult, Uint128};
+    use cosmwasm_std::{DepsMut, Env, MessageInfo, Response};
+
+    use cw20_base::contract::execute_send;
+    use cw20_base::contract::execute_transfer;
+
+    use cw20_base::allowances::execute_decrease_allowance;
+    use cw20_base::allowances::execute_increase_allowance;
+    use cw20_base::allowances::execute_send_from;
+    use cw20_base::allowances::execute_transfer_from;
+
+    use cw20_base::state::{BALANCES as RECEIPT_TOKEN_BALANCES, TOKEN_INFO as RECEIPT_TOKEN_INFO};
+
+    use crate::msg::ExecuteMsg as CombinedExecuteMsg;
+
+    /// This mint function is almost identical to the base cw20 contract's mint function
+    /// down to the variables and logic.
+    /// Except that it does not require the caller to be the minter.
+    pub fn mint_internal(
+        deps: DepsMut,
+        _env: Env,
+        _info: MessageInfo,
+        recipient: Addr,
+        amount: Uint128,
+    ) -> Result<Response, cw20_base::ContractError> {
+        let mut config = RECEIPT_TOKEN_INFO
+            .may_load(deps.storage)?
+            .ok_or(cw20_base::ContractError::Unauthorized {})?;
+
+        // update supply and enforce cap
+        config.total_supply += amount;
+        if let Some(limit) = config.get_cap() {
+            if config.total_supply > limit {
+                return Err(cw20_base::ContractError::CannotExceedCap {});
+            }
+        }
+        RECEIPT_TOKEN_INFO.save(deps.storage, &config)?;
+
+        // add amount to recipient balance
+        let rcpt_addr = recipient;
+        RECEIPT_TOKEN_BALANCES.update(
+            deps.storage,
+            &rcpt_addr,
+            |balance: Option<Uint128>| -> StdResult<_> { Ok(balance.unwrap_or_default() + amount) },
+        )?;
+
+        let res = Response::new()
+            .add_attribute("action", "mint")
+            .add_attribute("to", rcpt_addr)
+            .add_attribute("amount", amount);
+        Ok(res)
```

Comment:
```suggestion
        RECEIPT_TOKEN_BALANCES.update(
            deps.storage,
            &recipient,
            |balance: Option<Uint128>| -> StdResult<_> { Ok(balance.unwrap_or_default() + amount) },
        )?;

        Ok(Response::new()
            .add_attribute("action", "mint")
            .add_attribute("to", recipient)
            .add_attribute("amount", amount))
```

Identical code from upstream?

Anyway, `rcpt_addr` binding feels unnecessary, can remove the cost all together.